### PR TITLE
[FIX] web: filter date with locale

### PR DESCRIPTION
**Pull request by nboulif**
commit 1 : [IMP] web: utils; add util to merge objects accessors 
commit 2: [FIX] web: use user locale on Date(Time)Picker filter

### CODE REVIEW

**Comment by nboulif** [comment]
@kebeclibre Hello :slightly_smiling_face: First, thanks again :wink: I had an issue with the `Object.assign & Object.create` so I added a util to mix props (including getters) instead, what do you think...?
cc @Arcasias

**Comment by nboulif** [comment]
oups, forget tests...

**Comment by nboulif** [comment]
Test done :+1:

File: addons/web/static/src/js/core/utils.js [hunk 1]
```diff
@@ -1002,6 +1002,22 @@ var utils = {
             ['name', '=like', '%.assets\_%.js'],
         ];
     },
+    /**
+     * Merge multiple objects props ( including Getters and Setters )
+     * @param  {(Object|...Object)} sources
+     * @returns {Object} New object with sources objects props merged
+     */
+    mix : function (...sources) {
```
Comment (kebeclibre) [inline addons/web/static/src/js/core/utils.js:1010]: Let's see what @aab-odoo  would say,
In my opinion a goo'ol' object.create/object.assign is better, less magic, more straightforward than an obscure utility that we'd have to maintain

This is generally related to how we do class inheritance in a owl context
So pinging @mcm-odoo @ged-odoo ; just to think about it in the patch method PR

**Review by aab-odoo — APPROVED** [review]

**Comment by aab-odoo** [comment]
robodoo r+

**Pull request closed without merging**